substrate/frame/assets/src/lib.rs

```diff
@@ -1751,6 +1793,104 @@ pub mod pallet {
 			Self::deposit_event(Event::<T, I>::Blocked { asset_id: id, who });
 			Ok(())
 		}
+
+		/// Revoke irreversibly Owner, Issuer, Admin, and Freezer privileges of the asset. Freeze
+		/// the metadata.
+		///
+		/// Warning: this action is irreversible.
+		///
+		/// Origin must be either Signed and the sender should be the Owner of the asset `id` or
+		/// `ForceOrigin`.
+		///
+		/// If origin is the owner: The deposit of the asset details and metadata will be
+		/// unreserved then withdrawn and sent to `DepositDestinationOnRevocation`. If no metadata
+		/// was set then the new deposit for the new metadata will be withdrawn from the owner.
+		///
+		/// If origin is `ForceOrigin`: The deposit of owner is unreserved.
+		///
+		/// Asset must be in `Live` status.
+		///
+		/// - `id`: The identifier of the asset.
+		///
+		/// Emits `OwnerAndTeamRevoked` and `MetadataSet`.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(32)]
+		pub fn revoke_all_privileges(
+			origin: OriginFor<T>,
+			id: T::AssetIdParameter,
+		) -> DispatchResult {
+			let maybe_owner_origin = match T::ForceOrigin::try_origin(origin) {
+				Ok(_) => None,
+				Err(origin) => Some(ensure_signed(origin)?),
+			};
+			let id: T::AssetId = id.into();
+			let mut asset = Asset::<T, I>::get(id.clone()).ok_or(Error::<T, I>::Unknown)?;
+
+			if let Some(owner) = maybe_owner_origin.as_ref() {
+				ensure!(asset.owner() == Some(&owner), Error::<T, I>::NoPermission);
+			}
+			ensure!(asset.status == AssetStatus::Live, Error::<T, I>::IncorrectStatus);
+
+			let old_metadata = Metadata::<T, I>::get(&id);
+
+			let deposit_amount = asset.deposit.saturating_add(old_metadata.deposit);
+
+			let fee_imbalance = if let Some(owner) = maybe_owner_origin {
+				let new_metadata_deposit =
+					Self::calc_metadata_deposit(&old_metadata.name, &old_metadata.symbol);
+
+				// NOTE: if new deposit is less than old deposit then the excess is not refunded.
+				let metadata_deposit_diff =
+					new_metadata_deposit.saturating_sub(old_metadata.deposit);
+
+				with_storage_layer::<_, DispatchError, _>(|| {
+					T::Currency::unreserve(&owner, deposit_amount);
+					let imbalance = T::Currency::withdraw(
+						&owner,
+						deposit_amount.saturating_add(metadata_deposit_diff),
+						WithdrawReasons::FEE,
+						ExistenceRequirement::KeepAlive,
+					)?;
+					Ok(imbalance)
+				})?
+			} else if let Some(owner) = asset.owner() {
```

Comment:
the check is only made if call is from an owner_origin.
if the call is from `ForceOrigin` then the else if branch if used.

That is also why in case call is from `ForceOrigin` then the deposit is unreserved.
In case the call is from owner then the deposit for the old metadata is unreserved and the deposit for the new metadata is withdrawned from owner.